src/gui/pages_game/KM_GUIGameHouse.pas

```diff
@@ -831,7 +831,19 @@ procedure TKMGUIGameHouse.House_SchoolUnitChange(Sender: TObject; Shift: TShiftS
   if (Sender = Button_School_Right) and (fLastSchoolUnit < High(School_Order)) then Inc(fLastSchoolUnit);
 
   if Sender = Button_School_Train then //Add unit to training queue
-    gGame.GameInputProcess.CmdHouse(gic_HouseSchoolTrain, School, School_Order[fLastSchoolUnit], GetMultiplicator(Shift));
+  begin
+    if (ssRight in Shift) then
+      gGame.GameInputProcess.CmdHouse(gic_HouseSchoolTrain, School, School_Order[fLastSchoolUnit], 10)
+    else if (ssLeft in Shift) then
+    begin
+      gGame.GameInputProcess.CmdHouse(gic_HouseSchoolTrain, School, School_Order[fLastSchoolUnit], 1);
+      ChOrderFrom := max(School.LastUnitPosInQueue, 1);

```

Comment:
I suspect calculation of `ChOrderFrom` should be moved into GIP command processing, as `LastUnitPosInQueue` can change during lag between click and execution.